spec/integration/integration_spec_helper.rb

```diff
@@ -64,6 +65,48 @@ def normalize_inspect_output(output)
     gsub!(/(\r\033\[K.*?\r\033\[K).*\r\033\[K(.*)/, "\\1\\2") # strip all progress messages but two
 end
 
+def machinery_boxes
+  {
+    "openSUSE_13_2" => "machinery_132",
+    "openSUSE_13_1" => "machinery_131",
+    "Tumbleweed" => "machinery_tumbleweed"
+  }
+end
+
+def boxes
+  {
+    "openSUSE_13_2" => "opensuse132",
+    "openSUSE_13_1" => "opensuse131"
+  }
+end
+
+def test_according_to_matrix(currently_running_on)
+  path = File.join(Machinery::ROOT, "..", "alfred", "documents", "source", "integration_tests.yml")
+  matrix = YAML.load_file(path)
+
+  matrix.each do |command|
+    command_name, support_level = command
+    support_level.each do |_type, test_relations|
+      test_relations.each do |relation|
+        relation.each do |host, guests|
```

Comment:
I think that if you make the arrays in the YAML matrix hashes instead it will easier to iterate the matrix, e.g.

```
  inspect:
    supported:
      "sles12:x86_64":
        "sles12:x86_64": full_test
        "sles12:s390x": full_test
        "sles12:power_le": full_test
        "sles11:x86_64": full_test
        "rhel5": full_test
        "rhel6": full_test
```

you should be able to iterate with 

```
test_relations.each do |host, guests|
``` 

directly.